src/runtime/testing/callback-tracker.ts

```diff
@@ -0,0 +1,48 @@
+/**
+ * @license
+ * Copyright (c) 2017 Google Inc. All rights reserved.
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ * Code distributed by Google as part of this project is also
+ * subject to an additional IP rights grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+import {assert} from '../../platform/chai-web.js';
+import {StorageProviderBase} from '../storage/storage-provider-base.js';
+
+/**
+ * Simple class to verify callbacks used in the Arcs storage APIs.
+ *
+ * Usage:
+ * ```
+ *   const varCallbacks = new CallbackTracker(var1, 6);
+ *   .... // do work
+ *   varCallbacks.verify();
+ * ```
+ */
+export class CallbackTracker {
```

Comment:
this class is storage specific, so maybe rename to StorageCallbackTracker?

Reply:
I'd like to make it more generic so it can be used with any callback, but that's blocked on getting a generic mocking system in place.

Reply:
Added a TODO